### PR TITLE
HBASE-22975 Backport HBASE-22975 to branch-1

### PR DESCRIPTION
**Pull request by ZhaoBQ**
[HBASE-22975](https://issues.apache.org/jira/browse/HBASE-22975)

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   1m 26s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| :blue_heart: |  mvndep  |   5m 48s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   3m 35s |  branch-1 passed  |
| :green_heart: |  compile  |   2m 44s |  branch-1 passed with JDK v1.8.0_222  |
| :broken_heart: |  compile  |   0m 14s |  hbase-hadoop-compat in branch-1 failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 14s |  hbase-hadoop2-compat in branch-1 failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 22s |  hbase-server in branch-1 failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   2m 21s |  branch-1 passed  |
| :green_heart: |  shadedjars  |   3m 23s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m  2s |  branch-1 passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   1m 14s |  branch-1 passed with JDK v1.7.0_232  |
| :blue_heart: |  spotbugs  |   3m 18s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   4m 33s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 17s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   2m 16s |  the patch passed  |
| :green_heart: |  compile  |   2m 38s |  the patch passed with JDK v1.8.0_222  |
| :broken_heart: |  javac  |   0m 23s |  hbase-hadoop2-compat-jdk1.8.0_222 with JDK v1.8.0_222 generated 3 new + 36 unchanged - 3 fixed = 39 total (was 39)  |
| :broken_heart: |  compile  |   0m 13s |  hbase-hadoop-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 13s |  hbase-hadoop2-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 23s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 13s |  hbase-hadoop-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 13s |  hbase-hadoop2-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 23s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   2m 13s |  the patch passed  |
| :broken_heart: |  whitespace  |   0m  0s |  The patch has 1 line(s) that end in whitespace. Use git apply --whitespace=fix <<patch_file>>. Refer https://git-scm.com/docs/git-apply  |
| :green_heart: |  shadedjars  |   3m  8s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |   6m  4s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| :green_heart: |  javadoc  |   1m  9s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   1m 25s |  the patch passed with JDK v1.7.0_232  |
| :green_heart: |  findbugs  |   5m 15s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 28s |  hbase-hadoop-compat in the patch passed.  |
| :green_heart: |  unit  |   0m 37s |  hbase-hadoop2-compat in the patch passed.  |
| :broken_heart: |  unit  | 159m 17s |  hbase-server in the patch failed.  |
| :green_heart: |  asflicense  |   1m  7s |  The patch does not generate ASF License warnings.  |
|  |   | 218m 21s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestSnapshotCloneIndependence |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/648 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 203a8158df65 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-648/out/precommit/personality/provided.sh |
| git revision | branch-1 / dd761b1 |
| Default Java | 1.7.0_232 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_222 /usr/lib/jvm/zulu-7-amd64:1.7.0_232 |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/branch-compile-hbase-hadoop-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/branch-compile-hbase-hadoop2-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/branch-compile-hbase-server-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/diff-compile-javac-hbase-hadoop2-compat-jdk1.8.0_222.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-compile-hbase-hadoop-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-compile-hbase-hadoop2-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-compile-hbase-hadoop-compat-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-compile-hbase-hadoop2-compat-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/whitespace-eol.txt |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/testReport/ |
| Max. process+thread count | 4905 (vs. ulimit of 10000) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-hadoop2-compat/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsTableQueryMeterImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  @Override
+  public void updateTableWriteQueryMeter(TableName tableName, long count) {
+    getOrCreateTableMeter(tableName).updateTableWriteQueryMeter(count);
+  }
+
+  @Override
+  public void updateTableWriteQueryMeter(TableName tableName) {
+    getOrCreateTableMeter(tableName).updateTableWriteQueryMeter();
+  }
+}
```
Comment (Reidddddd) [inline hbase-hadoop2-compat/src/main/java/org/apache/hadoop/hbase/regionserver/MetricsTableQueryMeterImpl.java]: nit, an empty line at the end of file.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java [hunk 2]
```diff
@@ -5849,7 +5853,7 @@ public boolean bulkLoadHFiles(Collection<Pair<byte[], String>> familyPaths, bool
     boolean isSuccessful = false;
     try {
       this.writeRequestsCount.increment();
-
```
Comment (Reidddddd) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java]: Unnecessary change.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| :blue_heart: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| :yellow_heart: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| :blue_heart: |  mvndep  |   5m 45s |  Maven dependency ordering for branch  |
| :green_heart: |  mvninstall  |   3m 33s |  branch-1 passed  |
| :green_heart: |  compile  |   2m 41s |  branch-1 passed with JDK v1.8.0_222  |
| :broken_heart: |  compile  |   0m 13s |  hbase-hadoop-compat in branch-1 failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 14s |  hbase-hadoop2-compat in branch-1 failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 21s |  hbase-server in branch-1 failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   2m 18s |  branch-1 passed  |
| :green_heart: |  shadedjars  |   3m 20s |  branch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  javadoc  |   1m 10s |  branch-1 passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   1m 15s |  branch-1 passed with JDK v1.7.0_232  |
| :blue_heart: |  spotbugs  |   3m  1s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| :green_heart: |  findbugs  |   4m 12s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| :blue_heart: |  mvndep  |   0m 15s |  Maven dependency ordering for patch  |
| :green_heart: |  mvninstall  |   2m  8s |  the patch passed  |
| :green_heart: |  compile  |   2m 28s |  the patch passed with JDK v1.8.0_222  |
| :broken_heart: |  javac  |   0m 19s |  hbase-hadoop2-compat-jdk1.8.0_222 with JDK v1.8.0_222 generated 3 new + 36 unchanged - 3 fixed = 39 total (was 39)  |
| :broken_heart: |  compile  |   0m 13s |  hbase-hadoop-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 14s |  hbase-hadoop2-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  compile  |   0m 22s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 13s |  hbase-hadoop-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 14s |  hbase-hadoop2-compat in the patch failed with JDK v1.7.0_232.  |
| :broken_heart: |  javac  |   0m 22s |  hbase-server in the patch failed with JDK v1.7.0_232.  |
| :green_heart: |  checkstyle  |   2m 17s |  the patch passed  |
| :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| :green_heart: |  shadedjars  |   3m  7s |  patch has no errors when building our shaded downstream artifacts.  |
| :green_heart: |  hadoopcheck  |   5m 15s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| :green_heart: |  javadoc  |   0m 57s |  the patch passed with JDK v1.8.0_222  |
| :green_heart: |  javadoc  |   1m 14s |  the patch passed with JDK v1.7.0_232  |
| :green_heart: |  findbugs  |   5m 13s |  the patch passed  |
||| _ Other Tests _ |
| :green_heart: |  unit  |   0m 29s |  hbase-hadoop-compat in the patch passed.  |
| :green_heart: |  unit  |   0m 39s |  hbase-hadoop2-compat in the patch passed.  |
| :green_heart: |  unit  | 135m 46s |  hbase-server in the patch passed.  |
| :green_heart: |  asflicense  |   1m  1s |  The patch does not generate ASF License warnings.  |
|  |   | 191m 44s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/648 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d71243a20b3d 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-648/out/precommit/personality/provided.sh |
| git revision | branch-1 / 7239ccb |
| Default Java | 1.7.0_232 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_222 /usr/lib/jvm/zulu-7-amd64:1.7.0_232 |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/branch-compile-hbase-hadoop-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/branch-compile-hbase-hadoop2-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/branch-compile-hbase-server-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/diff-compile-javac-hbase-hadoop2-compat-jdk1.8.0_222.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/patch-compile-hbase-hadoop-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/patch-compile-hbase-hadoop2-compat-jdk1.7.0_232.txt |
| compile | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/patch-compile-hbase-hadoop-compat-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/patch-compile-hbase-hadoop2-compat-jdk1.7.0_232.txt |
| javac | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/artifact/out/patch-compile-hbase-server-jdk1.7.0_232.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/testReport/ |
| Max. process+thread count | 3987 (vs. ulimit of 10000) |
| modules | C: hbase-hadoop-compat hbase-hadoop2-compat hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-648/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.0 https://yetus.apache.org |


This message was automatically generated.
